### PR TITLE
Migrate IexecAccessors tests Typescript & Hardhat

### PR DESCRIPTION
**Pull request by zguesmi**

### CODE REVIEW

File: test/byContract/IexecAccessors/IexecAccessors.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { deployments, ethers, expect } from 'hardhat';
+import { loadHardhatFixtureDeployment } from '../../../scripts/hardhat-fixture-deployer';
+import { IexecInterfaceNative, IexecInterfaceNative__factory } from '../../../typechain';
+import { getIexecAccounts } from '../../../utils/poco-tools';
+
+/**
+ * Test constant view functions.
+ */
```
Comment (jeremyjams) [inline test/byContract/IexecAccessors/IexecAccessors.ts:13]: Can you remove admin & iexecPoco please?

**Review by gfournieriExec — APPROVED** [review]

File: test/byContract/IexecAccessors/IexecAccessors.ts (hunk 1 above)
Comment (zguesmi) [reply test/byContract/IexecAccessors/IexecAccessors.ts:13]: [Keep only pocoAsAnyone instance](https://github.com/iExecBlockchainComputing/PoCo/pull/96/commits/7d814b0c25a874d8e66430b55c909c7bd1febb6b)

File: test/byContract/IexecAccessors/IexecAccessors.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                (await deployments.get('DatasetRegistry')).address,
+            );
+        });
+        it('workerpoolRegistry', async function () {
+            expect(await iexecPocoAsAnyone.workerpoolregistry()).to.equal(
+                (await deployments.get('WorkerpoolRegistry')).address,
+            );
+        });
+    });
+});
```
[the remark this answers is not in the document]
Comment (zguesmi) [reply test/byContract/IexecAccessors/IexecAccessors.ts:104]: I updated the order to stick to the occurrence order in the source file. I find it clearer/cleaner that way.
 [Reorder tests according to source file functions order](https://github.com/iExecBlockchainComputing/PoCo/pull/96/commits/d5afde94674ab7d2f0cc4a18814bb064b0d15ef8)

File: test/byContract/IexecAccessors/IexecAccessors.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { deployments, ethers, expect } from 'hardhat';
+import { loadHardhatFixtureDeployment } from '../../../scripts/hardhat-fixture-deployer';
+import { IexecInterfaceNative, IexecInterfaceNative__factory } from '../../../typechain';
+import { getIexecAccounts } from '../../../utils/poco-tools';
+
+/**
+ * Test constant view functions.
```
[the remark this answers is not in the document]
Comment (jeremyjams) [reply test/byContract/IexecAccessors/IexecAccessors.ts:12]: Some are upgradable that's why I want to remove `constant` word!

File: test/byContract/IexecAccessors/IexecAccessors.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    async function initFixture() {
+        const accounts = await getIexecAccounts();
+        ({ anyone } = accounts);
+        iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, iexecAdmin);
+        iexecPocoAsAnyone = iexecPoco.connect(anyone);
+    }
+
+    describe('Config', function () {
+        it('token', async function () {
+            expect(await iexecPocoAsAnyone.token()).to.equal(
```
[the remark this answers is not in the document]
Comment (zguesmi) [reply test/byContract/IexecAccessors/IexecAccessors.ts]: Added a TODO for that.
[Add todo](https://github.com/iExecBlockchainComputing/PoCo/pull/96/commits/9386532df244617133090cb97c08b0dc1604ab94)

**Review by jeremyjams — APPROVED** [review]
Good thanks!

**Pull request merged**